arbitrum-docs/stylus/stylus-quickstart.md

```diff
@@ -11,209 +11,242 @@ import PublicPreviewBannerPartial from '../partials/_public-preview-banner-parti
+
+```shell
 Finished release [optimized] target(s) in 1.88s
 Reading WASM file at hello-stylus/target/wasm32-unknown-unknown/release/hello-stylus.wasm
 Compressed WASM size: 3 KB
 Program succeeded Stylus onchain activation checks with Stylus version: 1
 ```
 
-Note that running `cargo stylus check` may take a few minutes, especially on a fresh build. If you're running the check or deploy commands, a message at the start will inform you that this process might take some time. For faster checks, you can use the `--no-verify` option. Also, please ensure that Docker is up and running, as these commands are required to be executed properly.
+Note that running `cargo stylus check` may take a few minutes, especially if you're checking a contract for the first time. 
+You can use the `-no-verify` option for faster checks.
 
-Once you're ready to deploy your program onchain, you can use the `cargo stylus deploy` subcommand as follows. First, we can estimate the gas required to perform our deployment with:
+See `cargo stylus check --help` for more options.
 
-```
-cargo stylus deploy \
-  --private-key-path=<PRIVKEY_FILE_PATH> \
-  --estimate-gas-only
-```
+## Deploying your contract
 
-and see:
+Once you're ready to deploy your program onchain, `cargo stylus deploy` will help you with the estimation of the deployment's gas cost, as well as with the deployment itself.
 
+### Estimating gas
+
+
+First, you can estimate the gas required to perform your contract's deployment by running:
+
+```shell
+cargo stylus deploy \
+  --private-key-path=PRIV_KEY_PATH \
+  --estimate-gas
 ```
-Compressed WASM size: 3 KB
-Deploying program to address 0x457b1ba688e9854bdbed2f473f7510c476a3da09
-Estimated gas: 12756792
+The command should return something like this:
+
+```shell
+deployment tx gas: 7123737
+gas price: "0.100000000" gwei
+deployment tx total cost: "0.000712373700000000" ETH
 ```
+### Deployment
 
-Next, attempt an actual deployment. Two transactions will be sent onchain.
 
-```
+Next, you can attempt an actual deployment. Two transactions will be sent onchain:
+
+```shell
 cargo stylus deploy \
-  --private-key-path=<PRIVKEY_FILE_PATH>
+  --private-key-path=PRIV_KEY_PATH
 ```
 
-and see:
+Once the deployment is successful, you'll see an output similar to this:
 
+```shell
+deployed code at address: 0x33f54de59419570a9442e788f5dd5cf635b3c7ac
+deployment tx hash: 0xa55efc05c45efc63647dff5cc37ad328a47ba5555009d92ad4e297bf4864de36
+wasm already activated!
 ```
-Compressed WASM size: 3 KB
-Deploying program to address 0x457b1ba688e9854bdbed2f473f7510c476a3da09
-Estimated gas: 12756792
-Submitting tx...
-Confirmed tx 0x42db…7311, gas used 11657164
-Activating program at address 0x457b1ba688e9854bdbed2f473f7510c476a3da09
-Estimated gas: 14251759
-Submitting tx...
-Confirmed tx 0x0bdb…3307, gas used 14204908
 
-```
+More options are available for sending and outputting your transaction data. See `cargo stylus deploy --help` for more details.
```

Comment:
We should also add a few more steps to show how to interact with the contract. We can do that in a separate PR, but I think we need to have it here. This would also require to use Foundry, but it's a necessary tool for any developer.

Reply:
I intend to add more, but we'll have to be mindful not to bloat the quickstart. I'm considering adding sequel articles to gradually add new instructions.